frontend/app/(tabs)/home.tsx

```diff
@@ -72,14 +62,42 @@ const Index = () => {
     } finally {
       setIsLoading(false);
     }
-  }, [page, isLoading]);
+  }, [page, isLoading, searchText]);
+
+  const handleSearch = (text: string) => {
+    setSearchText(text);
+
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
+
+    searchTimeout.current = setTimeout(() => {
+      if (text.trim().length >= 2) {
+        loadInitialCards(text);
+      } else if (text.trim() === '') {
+        loadInitialCards();
+      }
+    }, 500);
+  };
+
+  const handleSearchSubmit = () => {
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
 
-  const handleSearch = (searchText: string) => {
-    setSearchText(searchText);
+    if (searchText.trim().length >= 2) {
+      loadInitialCards(searchText);
+    } else if (searchText.trim() === '') {
+      loadInitialCards();
+    }
   };
 
   const clearSearch = () => {
     setSearchText('');
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
+    loadInitialCards();
```

Comment:
let's remove the timeout, possibly we should cache the `loadInitialCards` request so that we don't refetch everytime we clear the search field. Let's do that later using react query